### ISSUE TITLE
Rule Change: validate `allow` option of `no-console`

### ISSUE THREAD

**Issue by Tanujkanti4441** [eslint/eslint#19353]
### What rule do you want to change?

[no-console](https://eslint.org/docs/latest/rules/no-console)

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

Other

### Example code

```js
/* eslint no-console: ["error", { allow: ["warn", "eror"] }] */

console.warn("Log a warn level message.");
console.error("Log an error level message."); // This will be reported
```

### What does the rule currently do for this code?

Right now rule doesn't validate the strings in `allow` option and a typo like `eror` is allowed.

### What will the rule do after it's changed?

only allowed `console` methods will be valid inside `allow` option's array.

### Participation

- [x] I am willing to submit a pull request to implement this change.

**Comment by Rec0iL99** [comment]
Hi @Tanujkanti4441, I think the `console` object can indeed contain different methods depending on the environment it is called in. For example, browsers (like Chrome) support `console.memory`, while Node.js does not. Similarly, there might be additional methods specific to other runtime environments.

Because of this variability, maintaining a static list of console methods for validation might not be ideal

**Comment by mdjermanovic** [comment]
I agree with @Rec0iL99. `console` is not defined by the ECMAScript Specification. It's environment-specific so maintaining a list of methods doesn't seem feasible.

**Comment by Tanujkanti4441** [comment]
Okk, that make sense, so its better to leave this idea for now.
